compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/converter/DeclarationsConverter.kt

```diff
@@ -837,7 +837,7 @@ class DeclarationsConverter(
         }
 
         val typeAliasName = identifier.nameAsSafeName()
-        return withChildClassName(typeAliasName) {
+        return withChildClassName(typeAliasName, context.firFunctionTargets.isNotEmpty()) {
```

Comment:
Would it work to make the default value of `isLocal` to be `firFunctionTargets.isNotEmpty()`? Would that work for the other places where you explicitly pass `false`? It's odd to have this logic outside of the `Context` class.

Reply:
> Would it work to make the default value of `isLocal` to be `firFunctionTargets.isNotEmpty()`? Would that work for the other places where you explicitly pass `false`?

Yes, it works. I still believe passing bits explicitly for enum and anonymous object make sense, but at least using the old way of using function targets as a default value made this change as minimal as possible.

> It's odd to have this logic outside of the `Context` class.

I think the whole purpose of `Context` is to make users of it "context-aware" (or "context-sensitive"), which means, it's up to call-site to tweak its logic, depending on under what context it is. So, the current data model makes perfect sense to me, and it's rather odd to have such _guideline_ of how to use context inside `Context`. Please note that `firFunctionTargets` are manually updated outside of `Context` when targetable functions are visited. (And there, again, anonymous initializer is (and should be) missed, which is the motivation of this PR.)

Some additional thoughts. In theory, `Context` should be also immutable so that any independent elements can be handled with different immutable, independent contexts. Well, maybe this mutable data model is also part of what `fir` gains performance against the old frontend, but I wonder, if we start trying to running things in parallel, those mutable data models will bother. Even in the `with` utils about context changes, I saw that, with incorrect manipulation, quite unexpected stack push/pop happened. (That's why I used `removeLast` instead of `pop`, which silently hides stack consistency.) Anyway, it's too early to think about this.

Reply:
> > Would it work to make the default value of `isLocal` to be `firFunctionTargets.isNotEmpty()`? Would that work for the other places where you explicitly pass `false`?
> 
> Yes, it works. I still believe passing bits explicitly for enum and anonymous object make sense, but at least using the old way of using function targets as a default value made this change as minimal as possible.

I agree, I only meant in the other places outside of anonymous object and enum. Current code looks good!